zebra-state/src/sled_state.rs

```diff
@@ -77,8 +78,15 @@ impl SledState {
         };
 
         while let Some(queued_block) = self.queued_by_prev_hash.remove(&tip_hash()) {
-            self.commit_finalized(queued_block)
+            self.commit_finalized(queued_block);
+            metrics::counter!("state.committed.block.count", 1);
         }
+        if let Some(block::Height(height)) = read_tip()
```

Comment:
```suggestion
        if let Some(block::Height(height)) = read_tip(&hash_by_height)
```

Reply:
Is there some way that we could avoid doubling the number of tip reads that we do on each insertion?

Probably the cost is minimal compared to the cost of the insertion transaction so I'm not sure it matters that much, especially if it would make the code much worse.

Maybe we could change
```
            self.commit_finalized(queued_block);
            metrics::counter!("state.committed.block.count", 1);
```
to
```
            let height = queued_block.block.coinbase_height().unwrap();
            self.commit_finalized(queued_block);
            metrics::counter!("state.committed.block.count", 1);
            metrics::gauge!("state.committed.block.height", height);
```
or something?

Reply:
Yes, that works.

I also removed a mostly-redundant metrics call.